p2p/backend-test-suite/src/block_announcement.rs

```diff
@@ -183,12 +188,19 @@ where
     sync1.subscribe(&[PubSubTopic::Blocks]).await.unwrap();
     sync2.subscribe(&[PubSubTopic::Blocks]).await.unwrap();
 
-    let txs = (0..200_000)
-        .map(|_| {
-            SignedTransaction::new(Transaction::new(0, vec![], vec![], 0).unwrap(), vec![])
-                .expect("invalid witness count")
-        })
-        .collect::<Vec<_>>();
+    let input = TxInput::new(config.genesis_block_id().into(), 0);
+    let mut signature = Vec::new();
+    signature.resize(3 * 1024 * 1024, 0);
```

Comment:
Please don't define mutable vectors then do operations on them when you can make them in one liners. Here's how you do this:

```
let signature = (0..3 * 1024 * 1024).into_iter().map(|_| 0).collect::<Vec<u8>>();
```

Rule of thumb: If there's `mut`, then maybe this can be fixed. Only use `mut` if there's absolutely no other way.

First best solution, no mut.
Second best solution, move mutations to a separate function and return something immutable.

Reply:
I agree with the magic number thing, let's do that.

But the range I provided is correct. If you want size `N`, then it's `(0..N)`: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=1fa00a4836d43b58e8f84b27b1161b12

Reply:
Updated, thanks for the hint.